schemas/testpilot.js

```diff
@@ -1,21 +1,40 @@
 "use strict";
 
-const Joi = require("joi");
+const Joi = require("joi-browser");
 const commonSchema = require("./commonSchema");
 
-const testpilotEvent = Joi.object().keys({
-  event: Joi.string().required(),
-  object: Joi.string().required(),
-  timestamp: Joi.number().integer().required()
-});
-
+// commonSchema keys not shown below:
+// - event_type, called "event" in testpilot docs, and "method" in
+//   testpilot-metrics docs;
+// - client_id, called "clientId" in the Telemetry ping.
 const schema = commonSchema.keys({
-  payload: Joi.object().required().keys({
-    events: Joi.array().required().length(1).items(testpilotEvent),
-    test: Joi.string().valid("@testpilot-addon").required(),
-    timestamp: Joi.number().integer().required(),
-    version: Joi.string().required()
-  })
-}).options({allowUnknown: true});
+  // Require that the commonSchema "topic" field always be "testpilot".
+  topic: Joi.string().valid("testpilot").required(),
+
+  // Additional "testpilot" event fields:
+  // object: what was affected by "event", e.g., a button.
+  object: Joi.string(),
+
+  // client_time: non-PII timestamp, integer number of seconds since Firefox
+  //              application start.
+  client_time: Joi.number().integer().required(),
+  // variants: an optional string (either a simple string, or serialized JSON)
+  //           that describes any A/B or multivariate tests currently running.
+  //           See the variant docs in the mozilla/testpilot repo for more.
+  variants: Joi.string(),
```

Comment:
Maybe a variant strings example would be good. A given experiment might have `button-test--green` and `button-test--red` variants running in a specific version. You could build a usage chart that just overlays all the pings for a given date, bucketed by variants. I think that could be auto-generated, and would answer the question of which variant led to greater engagement.

I'm guessing most experiments won't run more than 1 or 2 variants at a time, so this should probably be fine...and if they want to get fancier, they can plan to invest time in fancier graphs, too.

Comments/questions welcome :-)